gms/gossiper.cc

```diff
@@ -1393,10 +1393,13 @@ endpoint_state& gossiper::get_endpoint_state(inet_address ep) {
 }
 
 future<> gossiper::reset_endpoint_state_map() {
-    _unreachable_endpoints.clear();
-    _live_endpoints.clear();
-    co_await update_live_endpoints_version();
-    co_await container().invoke_on_all([] (gossiper& g) {
+    logger.debug("Resetting endpoint state map");
+    auto lock = lock_endpoint_update_semaphore();
+    auto version = _live_endpoints_version + 1;
```

Comment:
The version is updated before _live_endpoints is really updated. I think it is safer to always update the _live_endpoints before update the _live_endpoints_version.

Reply:
But here `_live_endpoints_version` is not changed, we just calculate the next value (under the lock).
The new value is set below, after clearing the live and unreachable endpoints sets.

Reply:
You are right. Sorry, I misread it.